presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsMetadata.java

```diff
@@ -0,0 +1,375 @@
+        }
+
+        if (!partitioningColumn.isPresent()) {
+            throw new IllegalArgumentException(String.format("Table %s does not exist", tableHandle.getTableName()));
+        }
+
+        ConnectorTableLayout layout = createLayoutForPartitioningColumn(tableHandle, partitioningColumn.get());
+        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+    }
+
+    @Override
+    public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle handle)
+    {
+        TpcdsTableLayoutHandle layout = checkType(handle, TpcdsTableLayoutHandle.class, "layout");
+
+        return getTableLayouts(session, layout.getTable(), Constraint.alwaysTrue(), Optional.empty())
+                .get(0)
+                .getTableLayout();
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        TpcdsTableHandle tpcdsTableHandle = checkType(tableHandle, TpcdsTableHandle.class, "tableHandle");
+
+        Table table = Table.getTable(tpcdsTableHandle.getTableName());
+        String schemaName = scaleFactorSchemaName(tpcdsTableHandle.getScaleFactor());
+
+        return getTableMetadata(schemaName, table);
+    }
+
+    private static ConnectorTableMetadata getTableMetadata(String schemaName, Table tpcdsTable)
+    {
+        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+        for (Column column : tpcdsTable.getColumns()) {
+            columns.add(new ColumnMetadata(column.getName(), getPrestoType(column.getType())));
+    }
+        SchemaTableName tableName = new SchemaTableName(schemaName, tpcdsTable.getName());
+        return new ConnectorTableMetadata(tableName, columns.build());
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        ImmutableMap.Builder<String, ColumnHandle> builder = ImmutableMap.builder();
+        for (ColumnMetadata columnMetadata : getTableMetadata(session, tableHandle).getColumns()) {
+            builder.put(columnMetadata.getName(), new TpcdsColumnHandle(columnMetadata.getName(), columnMetadata.getType()));
+        }
+        return builder.build();
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, tableHandle);
+        String columnName = checkType(columnHandle, TpcdsColumnHandle.class, "columnHandle").getColumnName();
+
+        for (ColumnMetadata column : tableMetadata.getColumns()) {
+            if (column.getName().equals(columnName)) {
+                return column;
+            }
+        }
+        throw new IllegalArgumentException(String.format("Table %s does not have column %s", tableMetadata.getTable(), columnName));
```

Comment:
`TpchMetadata` uses `IllegalArgumentException` as well. I think it makes more semantic sense than `PrestoException`. I think the advantage of a `PrestoException` is the ability to classify errors based on their error codes but this kind of error probably won't happen often (as opposed to say HDFS timeouts where you want to keep track of those).

I also static imported `format()`.